src/tr1d1um/communication.go

```diff
@@ -50,8 +54,24 @@ func (tr1 *Tr1SendAndHandle) Send(ch *ConversionHandler, resp http.ResponseWrite
 	//todo: any more headers to be added here
 	requestToServer.Header.Set("Content-Type", wrp.JSON.ContentType())
 	requestToServer.Header.Set("Authorization", req.Header.Get("Authorization"))
-	respFromServer, err = tr1.timedClient.Do(requestToServer)
-	return
+
+	ctx, cancel := context.WithTimeout(context.Background(), tr1.timeout)
```

Comment:
When using requests, the typical way is to do `context.WithTimeout(request.Context(), tr1.timeout)`.  That makes it clear that you're augmenting the request's timeout.

If no context is associated with the request, `request.Context()` returns `context.Background()`.  See the godoc.